### PR TITLE
kubectl: apply prune should fallback to basic delete when a resource has no reaper

### LINKED ISSUE #34790: Need to implement reaper for PersistentVolume and PersistentVolumeClaims.
From #34513.

Experimenting the new feature `kubectl apply --prune`, found it unable to prune resource type `PersistentVolume` and `PersistentVolumeClaims`. Error as below:
```
$ kubectl apply -f pv-test.yaml --prune -l test="true"
...
error: error pruning objects: no reaper has been implemented for { PersistentVolume}
error: error pruning objects: no reaper has been implemented for { PersistentVolumeClaim}
```

We will need this for upgrading Addon Manager. It's weird that these resource types are not included. Or am I missing something?

@mikedanese @ymqytw

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #34790

cc @kubernetes/kubectl @MrHohn

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34836)

### CODE REVIEW

**Comment by smarterclayton** [comment]
LGTM, but is there a way to add a test?

**Comment by mikedanese** [comment]
Ya, I'll add a test.

**Comment by mikedanese** [comment]
Added a test.

File: hack/make-rules/test-cmd.sh [hunk 1]
```diff
@@ -1111,6 +1111,13 @@ __EOF__
   kube::test::get_object_assert 'pods b' "{{${id_field}}}" 'b'
   kubectl delete pod/a pod/b
 
+  ## kubectl apply --prune should fallback to delete for non reapable types
+  kubectl apply --all --prune -f hack/testdata/prune-reap/a.yml 2>&1 "${kube_flags[@]}"
+  kube::test::get_object_assert 'pvc a-pvc' "{{${id_field}}}" 'a-pvc'
+  kubectl apply --all --prune -f hack/testdata/prune-reap/b.yml 2>&1 "${kube_flags[@]}"
+  kube::test::get_object_assert 'pvc b-pvc' "{{${id_field}}}" 'b-pvc'
+  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+
```
Comment (mikedanese) [inline hack/make-rules/test-cmd.sh]: Done.

**Comment by MrHohn** [comment]
Thanks! LGTM

PS: I don't have the privilege to apply label.

File: pkg/kubectl/cmd/apply.go [hunk 2]
```diff
@@ -343,18 +343,18 @@ func (p *pruner) prune(namespace string, mapping *meta.RESTMapping, shortOutput
...
 	if err != nil {
 		return err
 	}
 	r, err := kubectl.ReaperFor(mapping.GroupVersionKind.GroupKind(), cs)
 	if err != nil {
-		return err
+		if _, ok := err.(*kubectl.NoSuchReaperError); !ok {
+			return err
+		}
+		return resource.NewHelper(c, mapping).Delete(namespace, name)
```
Comment (janetkuo) [inline pkg/kubectl/cmd/apply.go]: Optional but maybe add a warning here for falling back to non cascading deletion?

**Pull request merged**